src/codegen/function.rs

```diff
@@ -86,6 +90,54 @@ pub fn generate(
         }
     }
 
+    if analysis.async_future.is_some() {
+        let declaration = declaration_futures(env, analysis);
+        let suffix = if only_declaration { ";" } else { " {" };
+
+        try!(writeln!(w, ""));
```

Comment:
Please `try!(writeln!(w));` and in other 3 cases here

Reply:
done